src/main/scala/Battle.scala

```diff
@@ -0,0 +1,136 @@
+import scala.util.Random
+import scala.io.StdIn.{readLine,readInt}
+
+class Battle{
+  val r: Random.type = scala.util.Random
+  def action(){
+    println(s"Player health:${Player.health}\nPlayer attacks strength:${Player.attacks}\nEnemy health:${Enemy.health}\nEnemy strength:${Enemy.attacks}")
+    println("\nAction menu\n1. Attack\n2. Defend\n3. Spells \n4.Run away")
+    val choice = readInt
+    choice match {
+      case 1 => this.attack()
+      case 2 => this.defend()
+      case 3 => this.spells()
+      case 4 => println("TACTICAL RETREAT")
+      case _ => this.action()
+    }
+  }
+  def attack(){
+    println(s"\nYour initiative${Player.initiative}\nEnemy initiative${Enemy.initiative}")
+    if(Player.initiative > Enemy.initiative){
+      println(s"\nPlayer attacks strength:${Player.attacks}\nHit ratio:${Player.hit}")
+      println(s"\nEnemy attack strength:${Enemy.attacks}\nHit ratio:${Enemy.hit}")
+      if(Player.hit > Enemy.hit){
+        println("Player attack lands")
+        Enemy.enHealth = Enemy.health - Player.attacks
+        Enemy.health = Enemy.enHealth
+        println(s"Enemy health:${Enemy.enHealth}")
+        if(Enemy.health <= 0 || Enemy.enHealth <= 0){println("\nYOU WIN")}
+      }else{println("attack misses")}
+    }else{
+      println(s"\nEnemy attacks strength${Enemy.attacks}\nHit ratio${Enemy.hit}")
+      if(Enemy.hit > Player.hit){
+        println("Enemy lands hit")
+        val dmg = Player.armour + Player.bonusArmour - Enemy.attacks
+        if(dmg > 0){
+          println(s"You took $dmg damage ")
+          Player.health - dmg
+        }else{println("You took no damage")}
+        println(s"health:${Player.health}")
+        if(Player.health <= 0){println("\nYOU LOSE")}
+      }else{println("Enemy misses")}
+    }
+    if(Player.health <= 0){
+      println("DEAD")
+    }else{
+      println("\nContinue fighting (yes or no)")
+      val choice = readLine
+      if(choice == "y"|| choice == "Y" || choice == "yes" || choice == "Yes" || choice == "YES"){
+        Player.Reroll()
+        Enemy.Reroll()
+        this.action()
+      }else{println("\nYOU RUN AWAY")}
+    }
+  }
+  def defend(){
+    Player.bonusArmour = r.nextInt(5)
+    println(s"You gained bonus armour:${Player.bonusArmour}")
+    this.action()
+  }
+  def spells(){
```

Comment:
this is way too complex for what it should be, similarly the methods inside the "spells" are too big and also complex, another thing is you have quite a bit of duplicate code